### PR TITLE
Option to specify number of leading zeros in solution filename number

### PR DESCRIPTION
**Pull request by eytanadler**
## Purpose

In the existing version, the output file numbering is hard coded to be three digits long (e.g. 003). If there are more than 999 output files, the number is extended to four digits. This messes with file sorting in some cases because the 1001th case can come between the 101st and 102nd, depending on the sorting algorithm.

This PR adds a new option, `writeSolutionDigits`, to manually specify the number of digits in the output filename to get around this sorting problem.

## Expected time until merged

A few days

## Type of change

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (non-backwards-compatible fix or feature)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Documentation update
- [ ] Maintenance update
- [ ] Other (please describe)

## Testing

Run a case that writes out solution files. Try changing the `writeSolutionDigits` option to something other than the default of 3 and observe that the output filenames change.

## Checklist

- [x] I have run `flake8` and `black` to make sure the Python code adheres to PEP-8 and is consistently formatted
- [x] I have formatted the Fortran code with `fprettify` or C/C++ code with `clang-format` as applicable
- [ ] I have run unit and regression tests which pass locally with my changes
- [ ] I have added new tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation

### CODE REVIEW

**Review by sseraj — COMMENTED** [review]
Can you also update the failed mesh solution counter?

**Comment by eytanadler** [comment]
> Can you also update the failed mesh solution counter?

Yes chef!

**Review by lamkina — APPROVED** [review]
Love me a simple quality of life improvement!

**Review by sseraj — APPROVED** [review]
I tested that this works as intended

File: adflow/pyADflow.py [hunk 1]
```diff
@@ -1305,7 +1305,8 @@ def __call__(self, aeroProblem, **kwargs):
                 )
 
         if self.adflow.killsignals.fatalfail:
-            fileName = f"failed_mesh_{self.curAP.name}_{self.curAP.adflowData.callCounter:03}.cgns"
+            numDigits = self.getOption("writeSolutionDigits")
+            fileName = f"failed_mesh_{self.curAP.name}_{self.curAP.adflowData.callCounter:0{numDigits}}.cgns"
```
Comment (eirikurj) [inline adflow/pyADflow.py:1309]: Although it should be an int, we should probably add a trailing "d".

File: adflow/pyADflow.py [hunk 2]
```diff
@@ -2738,14 +2739,15 @@ def writeSolution(self, outputDir=None, baseName=None, number=None, writeSlices=
 
         # If we are numbering solution, it saving the sequence of
         # calls, add the call number
+        numDigits = self.getOption("writeSolutionDigits")
         if number is not None:
             # We need number based on the provided number:
-            baseName = baseName + "_%3.3d" % number
+            baseName = baseName + f"_%.{numDigits}d" % number
```
Comment (eirikurj) [inline adflow/pyADflow.py:2745]: Mixing c-style and f-string formatting looks messy. We should do either or, favoring the new f-string (like done above in this PR)

**Review by eirikurj — COMMENTED** [review]
Little late since this has been merged, but commenting mostly for documentation purposes. See the minor comments.

**Comment by eytanadler** [comment]
@eirikurj if you're ok mixing PRs, I'm happy to make these minor changes in #343.

**Pull request merged**